src/utils/frame.js

```diff
@@ -213,57 +213,60 @@ export function formatCopyName(frame: LocalFrame) {
   return `${displayName} (${fileName}#${frameLocation})`;
 }
 
-export function collapseFrames(frames: Frame[]) {
-  // We collapse groups of one so that user frames
-  // are not in a group of one
-  function addGroupToList(group, list) {
-    if (!group) {
-      return list;
-    }
-
-    if (group.length > 1) {
-      list.push(group);
-    } else {
-      list = list.concat(group);
-    }
-
-    return list;
-  }
-  const { newFrames, lastGroup } = collapseLastFrames(frames);
-  frames = newFrames;
-  let items = [];
-  let currentGroup = null;
-  let prevItem = null;
-  for (const frame of frames) {
-    const prevLibrary = get(prevItem, "library");
+export const collapseFrames: (Frame[]) => Array<Frame | Frame[]> = flow(
+  collapseLastWebpackFrames,
+  collapseBabelAsyncFrames,
+  collapseRemainingFrames
+);
 
-    if (!currentGroup) {
-      currentGroup = [frame];
-    } else if (prevLibrary && prevLibrary == frame.library) {
-      currentGroup.push(frame);
-    } else {
-      items = addGroupToList(currentGroup, items);
-      currentGroup = [frame];
-    }
-
-    prevItem = frame;
+function collapseLastWebpackFrames(frames: Frame[]) {
```

Comment:
What is this function doing?

Reply:
This is pre-existing functionality that I've put into a separate function.  Sometimes ([here](https://github.com/devtools-html/debugger.html/pull/5385/files#diff-7a84ca451c1af01a43cf706fa6736146R159), for example) you want the bottom frame to be considered a "webpack" frame even if the sourcemap URL doesn't identify it as such. @jasonLaster Did I get that right?